### PR TITLE
Manually inline the createSuper helper on older Babel versions

### PR DESCRIPTION
**Pull request by nicolo-ribaudo**
| Q                        | A 
| ------------------------ | ---
| Fixed Issues?            | Yes
| Patch: Bug Fix?          |
| Major: Breaking Change?  |
| Minor: New Feature?      |
| Tests Added + Pass?      | Yes
| Documentation PR Link    | 
| Any Dependency Changes?  |
| License                  | MIT

1. This feels hacky
2. I love this approach: helpers should live in the plugins, not in `@babel/core`. Alternatively, plugins should have a direct dependency on `@babel/helpers` (it might be easier).

### CODE REVIEW

File: packages/babel-plugin-transform-classes/src/inline-createSuper-helpers.js [hunk 1]
```diff
@@ -0,0 +1,71 @@
+import { template, types as t } from "@babel/core";
+
+const helperIDs = new WeakMap();
+
+export default function addCreateSuperHelper(file) {
+  try {
+    return file.addHelper("createSuper");
```
Comment (nicolo-ribaudo) [inline packages/babel-plugin-transform-classes/src/inline-createSuper-helpers.js]: I tested this PR by commenting out this line: all the `exec` tests passed, and the new `output`s made sense. I suggest trying it locally.

If you have an idea on how to better test this, please tell me!

File: packages/babel-plugin-transform-classes/src/inline-createSuper-helpers.js [hunk 2]
```diff
@@ -0,0 +1,71 @@
...
+    return (t.cloneNode || t.clone)(helperIDs.get(file));
+  }
+
+  const id = file.scope.generateUidIdentifier("createSuper");
+  helperIDs.set(file, id);
+
+  const fn = helper({
+    CREATE_SUPER: id,
+    GET_PROTOTYPE_OF: file.addHelper("getPrototypeOf"),
+    POSSIBLE_CONSTRUCTOR_RETURN: file.addHelper("possibleConstructorReturn"),
```
Comment (nicolo-ribaudo) [inline packages/babel-plugin-transform-classes/src/inline-createSuper-helpers.js]: Those 2 helpers are available starting from 7.0.0.

**Comment by nicolo-ribaudo** [comment]
@ianschmitz FYI, this PR (which we'll release today) fixes a bug reported in 2-3 comments in that long CRA issue (also, thanks for locking it: if you need anything from us feel free to ask!).

**Comment by ianschmitz** [comment]
Okay perfect. Thank you @nicolo-ribaudo. Can you give us the heads up when it's all released so we can bump the versions from our side? May as well fix it all in one shot.

You can contact me on twitter at https://twitter.com/schmitz_ian if that's easiest.

File: packages/babel-plugin-transform-classes/src/inline-createSuper-helpers.js [hunk 3]
```diff
@@ -0,0 +1,71 @@
...
+
+export default function addCreateSuperHelper(file) {
+  try {
+    return file.addHelper("createSuper");
+  } catch {
+    // Babel <7.9.0 doesn't support the helper.
+  }
+
+  if (helperIDs.has(file)) {
+    // t.cloneNode isn't supported in every version
```
Comment (JLHwung) [inline packages/babel-plugin-transform-classes/src/inline-createSuper-helpers.js]: nit: may leave a note here that we should drop `t.clone` in Babel 8.

File: packages/babel-plugin-transform-classes/src/inline-createSuper-helpers.js [hunk 4]
```diff
@@ -0,0 +1,71 @@
...
+    POSSIBLE_CONSTRUCTOR_RETURN: file.addHelper("possibleConstructorReturn"),
+  });
+
+  file.path.unshiftContainer("body", [fn]);
+  file.scope.registerDeclaration(file.path.get("body.0"));
+
+  return t.cloneNode(id);
+}
+
+const helper = template.statement`
```
Comment (JLHwung) [inline packages/babel-plugin-transform-classes/src/inline-createSuper-helpers.js]: May worth leave a note at `babel-helpers` that we should manually sync these two sources.

**Comment by loopmode** [comment]
See https://github.com/zeit/next.js/discussions/11233#discussioncomment-1739

**Comment by nicolo-ribaudo** [comment]
I'm releasing this hotfix, and also working on a proper fix to prevent this from happening again in the future.

**Pull request merged**